### ISSUE TITLE
Error.razor should always be a static server rendered page

### ISSUE THREAD

**Issue by danroth27** [dotnet/aspnetcore#51031]
If you create new Blazor Web App with interactive WebAssembly rendering enabled globally, then the Error.razor page will get included in the client project. But this doesn't really make sense. The error page uses the HttpContext, which isn't available on the client. The error page should always be a statically rendered page from the server.

**Comment by danroth27** [comment]
I see that this was discussed by @SteveSandersonMS here: https://github.com/dotnet/aspnetcore/pull/50920/files#r1336192873

**Comment by danroth27** [comment]
I tried moving Error.razor to the server project and then setting the render mode of the router dynamically based on the file path:

```razor
<body>
    <Routes @rendermode="@RenderModeForPage" />
    <script src="_framework/blazor.web.js"></script>
</body>

</html>

@code {
    [CascadingParameter] HttpContext HttpContext { get; set; } = default!;

    IComponentRenderMode? RenderModeForPage => HttpContext.Request.Path.StartsWithSegments("/error")
        ? null
        : RenderMode.InteractiveWebAssembly;
}
```

But then I get an `ArgumentNullException` when trying to set the render mode to `null`:

```
System.ArgumentNullException: Value cannot be null. (Parameter 'renderMode')
   at System.ArgumentNullException.Throw(String paramName)
   at System.ArgumentNullException.ThrowIfNull(Object argument, String paramName)
   at Microsoft.AspNetCore.Components.Rendering.RenderTreeBuilder.AddComponentRenderMode(IComponentRenderMode renderMode)
   at BlazorApp7.Components.App.BuildRenderTree(RenderTreeBuilder __builder) in C:\Users\daroth\source\repos\BlazorApp7\BlazorApp7\Components\App.razor:line 16
   at Microsoft.AspNetCore.Components.Rendering.ComponentState.RenderIntoBatch(RenderBatchBuilder batchBuilder, RenderFragment renderFragment, Exception& renderFragmentException)
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.ProcessRenderQueue()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.ProcessRenderQueue()
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.AddToRenderQueue(Int32 componentId, RenderFragment renderFragment)
   at Microsoft.AspNetCore.Components.ComponentBase.StateHasChanged()
   at Microsoft.AspNetCore.Components.ComponentBase.CallOnParametersSetAsync()
   at Microsoft.AspNetCore.Components.ComponentBase.RunInitAndSetParametersAsync()
   at Microsoft.AspNetCore.Components.Rendering.ComponentState.SetDirectParameters(ParameterView parameters)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.RenderRootComponentAsync(Int32 componentId, ParameterView initialParameters)
   at Microsoft.AspNetCore.Components.HtmlRendering.Infrastructure.StaticHtmlRenderer.BeginRenderingComponent(IComponent component, ParameterView initialParameters)
   at Microsoft.AspNetCore.Components.Endpoints.EndpointHtmlRenderer.RenderEndpointComponent(HttpContext httpContext, Type rootComponentType, ParameterView parameters, Boolean waitForQuiescence)
   at Microsoft.AspNetCore.Components.Endpoints.RazorComponentEndpointInvoker.RenderComponentCore(HttpContext context)
   at Microsoft.AspNetCore.Components.Endpoints.RazorComponentEndpointInvoker.RenderComponentCore(HttpContext context)
   at Microsoft.AspNetCore.Components.Rendering.RendererSynchronizationContext.<>c.<<InvokeAsync>b__10_0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
```

It looks we might need to change the runtime to provide a way to set the render mode to static server rendering.

**Comment by SteveSandersonMS** [comment]
> 
![Uploading image.png…]()

**Comment by halter73** [comment]
This was fixed by #51134
